src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -823,6 +832,28 @@ private void WriteProjectItems(IEnumerable items)
             }
         }
 
+        private const string EmbedInBinlogItemType = "EmbedInBinlog";
+
+        private void CheckForFilesToEmbed(string itemType, object itemList)
+        {
+            if (!string.Equals(itemType, EmbedInBinlogItemType, StringComparison.OrdinalIgnoreCase) || itemList is not IEnumerable list)
+            {
+                return;
+            }
+
+            foreach (var item in list)
+            {
+                if (item is ITaskItem taskItem && !string.IsNullOrEmpty(taskItem.ItemSpec))
```

Comment:
What if there itemspecs are plain strings that do not refer to files, or they are paths but the paths do not exist? Should you error / warn on these?

Reply:
they will be silently skipped, I think this is acceptable